core/Cargo.toml

```diff
@@ -19,18 +19,28 @@ doctest = false
 
 [features]
 default = ["std"]
+diagnostics = ["dep:miette"]
 std = [
-	"miden-crypto/std",
-	"miden-formatting/std",
-	"math/std",
-	"winter-utils/std",
-	"thiserror/std",
+    "dep:parking_lot",
+    "memchr/std",
+    "miden-crypto/std",
+    "miden-formatting/std",
+    "math/std",
+    "winter-utils/std",
+    "thiserror/std",
 ]
 
 [dependencies]
 math = { package = "winter-math", version = "0.9", default-features = false }
+memchr = { version = "2.7.4", default-features = false }
 miden-crypto = { git = "https://github.com/0xPolygonMiden/crypto", branch = "next", default-features = false }
 miden-formatting = { version = "0.1", default-features = false }
+miette = { version = "7.1.0", git = "https://github.com/bitwalker/miette", branch = "no-std", default-features = false, features = [
+    "fancy-no-syscall",
+    "derive",
+], optional = true }
+lock_api = { version = "0.4.12", features = ["arc_lock"] }
+parking_lot = { version = "0.12.3", optional = true }
```

Comment:
If the multi-threaded source manager moves to the compiler, we can get rid of these dependencies, right?

Reply:
For now? Probably. We may end up needing a multi-threaded source code manager though for debugging, at which point we have to do all of this again.